provider/internal/builtin/provider.go

```diff
@@ -25,8 +25,9 @@ type builtinCondition struct {
 }
 
 type fileContentCondition struct {
-	FilePattern string `yaml:"filePattern" json:"filePattern,omitempty" title:"FilePattern" description:"Only search in files with names matching this pattern"`
-	Pattern     string `yaml:"pattern" json:"pattern" title:"Pattern" description:"Regex pattern to match in content"`
+	FilePattern string   `yaml:"filePattern" json:"filePattern,omitempty" title:"FilePattern" description:"Only search in files with names matching this pattern"`
+	Pattern     string   `yaml:"pattern" json:"pattern" title:"Pattern" description:"Regex pattern to match in content"`
+	Filepaths   []string `yaml:"filepaths" json:"filepaths,omitempty" title:"Filepaths" description:"Optional list of files to scope down search"`
```

Comment:
Should we make this a helper struct in provider lib so that every referenced condition could just embed it to keep it consistant between everyone?